### PR TITLE
enhance: Improve the a11y of the upload progress bar

### LINKED ISSUE #42703: Progress is not announced correctly in Upload progress bar
### Reproduction link

[https://ant.design/components/upload#components-upload-demo-defaultfilelist](https://ant.design/components/upload#components-upload-demo-defaultfilelist)

### Steps to reproduce

1. Open the "Default Files" example in a new window
2. Open VoiceOver
3. Go to the first item in the file list that has a progress bar

### What is expected?

The element should announce the progress and the associated label

### What is actually happening?

The element is announced as "indeterminate, progress indicator"

| Environment | Info |
| --- | --- |
| antd | 5.5.1 |
| React | 18.0.0 |
| System | MacOS Ventura 13.3.1 |
| Browser | Edge 113.0.1774.57 |

Ref: UIEN-3474

### PR DESCRIPTION
**Pull request by MehmetYararVX**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [x] Other (about what?): Accessibility

### 🔗 Related issue link

fix #42703 

### 💡 Background and solution

The following accessibility features are added to the upload progress:

- `aria-valuenow` which announces the current progress percent
- `aria-label` which announced the name of the progress
- `aria-labelledby` which announces the associated label of the progress

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Improve Progress accessibility by adding more aria props.     |
| 🇨🇳 Chinese |  增强 Progress 可访问性表现。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 2d941d9</samp>

Improved the accessibility of the `Progress` and `Upload` components by adding new props for custom labels. Refactored some types and imports for clarity and consistency.

### <samp>🤖 Generated by Copilot at 2d941d9</samp>

*  Add `ariaLabel` and `ariaLabelledBy` props to `Progress` and `UploadFile` interfaces to allow accessible labels for progress and upload components ([link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddR55-R56), [link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-160ee6918ffebe6609f946497b4f3cb4dac8d6676f9c1eea3ba82e5ddc42aea8R38-R39))
*  Pass the new props from `props` and `file` to the `div` and `Progress` elements in `Progress` and `ListItem` functions respectively ([link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddR73-R74), [link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddR182-R184), [link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-958fc631d437799c784b9bd5b685e7e31d2a365e4eb2a3d8424c970e8d037862L250-R256))
*  Change the type annotation for `ProgressType` and `ProgressStatuses` to use brackets instead of `typeof` directly, to avoid confusion with the TypeScript operator ([link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL18-R18), [link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL39-R39), [link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL82-R86))
*  Reorder the import of `RcUploadProps` in `interface.ts` to match the alphabetical order of the other imported types ([link](https://github.com/ant-design/ant-design/pull/42704/files?diff=unified&w=0#diff-160ee6918ffebe6609f946497b4f3cb4dac8d6676f9c1eea3ba82e5ddc42aea8L3-R4))

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

**Comment by zombieJ** [comment]
CI failed. Use `npm test -- -u` to fix the snapshot.

**Comment by afc163** [comment]
Is it better to use https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress instead of div.

File: components/progress/progress.tsx [hunk 1]
```diff
@@ -68,6 +70,8 @@ const Progress = React.forwardRef<HTMLDivElement, ProgressProps>((props, ref) =>
     type = 'line',
     status,
     format,
+    ariaLabel,
+    ariaLabelledBy,
```
Comment (afc163) [inline components/progress/progress.tsx:74]: We could remove it, these are included in `restProps`.

File: components/progress/__tests__/index.test.tsx [hunk 2]
```diff
@@ -385,4 +385,23 @@ describe('Progress', () => {
     );
     errSpy.mockRestore();
   });
+
+  it('should be accessible', () => {
+    const { container: wrapper, rerender } = render(
+      <Progress percent={70} ariaLabel="My progress" />,
```
Comment (afc163) [inline components/progress/__tests__/index.test.tsx:391]:
```suggestion
      <Progress percent={70} aria-label="My progress" />,
```

https://legacy.reactjs.org/docs/accessibility.html#wai-aria

File: components/upload/interface.ts [hunk 3]
```diff
@@ -35,6 +35,8 @@ export interface UploadFile<T = any> {
   type?: string;
   xhr?: T;
   preview?: string;
+  ariaLabel?: string;
+  ariaLabelledBy?: string;
```
Comment (afc163) [inline components/upload/interface.ts:39]: https://legacy.reactjs.org/docs/accessibility.html#wai-aria

File: components/progress/progress.tsx [hunk 4]
```diff
@@ -52,6 +52,8 @@ export interface ProgressProps {
   steps?: number;
   /** @deprecated Use `success` instead */
   successPercent?: number;
+  ariaLabel?: string;
+  ariaLabelledBy?: string;
```
Comment (afc163) [inline components/progress/progress.tsx:56]: https://legacy.reactjs.org/docs/accessibility.html#wai-aria

File: components/progress/progress.tsx [hunk 5]
```diff
@@ -52,6 +52,8 @@ export interface ProgressProps {
...
   children?: React.ReactNode;
 }
```
Comment (afc163) [inline components/progress/progress.tsx:58]:
```suggestion
} & React.AriaAttributes
```

Comment (afc163) [reply components/progress/progress.tsx:58]: Try this way.

**Pull request merged**